src/quest_manager/models.py

```diff
@@ -975,3 +989,150 @@ def get_minutes_to_complete(self):
+            "specific guidance, a prompt, or a problem statement for the student to address."
+        )
+    )
+    solution_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, allowing for an easy comparison of "
+            "the student's submission to an ideal or example solution."
+        ),
+    )
+    solution_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/solution/%Y/%m/%d",
+        help_text=(
+            "This file will appear to the marker, allowing them to easily compare "
+            "the student's submission to an ideal or example solution for file-based questions."
+        ),
+    )
+
+    # Each tuple contains the file type as both the stored value and the display value
+    # e.g. ("media", "Media") or ("image", "Image")
+    allowed_file_type = models.CharField(
+        max_length=50,
+        choices=[(key, key.capitalize()) for key in FILE_MIME_TYPES.keys()],
+        default="media",
+        help_text=(
+            "The types of files that can be uploaded by a student for this question."
+        ),
+    )
+
+    marker_notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, providing additional "
+            "context and info, if needed."
+        ),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=(
+            "If this field is unchecked, the student can complete the quest without "
+            "answering this question."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Questions will be displayed to the student in order from lowest ordinal to "
+            "highest ordinal."
+        ),
+    )
+
+    quest = models.ForeignKey(Quest, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=20, choices=QuestionType.choices, default=QuestionType.SHORT_ANSWER
+    )
+
+    class Meta:
+        ordering = ["ordinal"]
+        # ensure two questions in the same quest cannot have the same ordinal
+        unique_together = ["quest", "ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal])
+        e.g. "Quest 1 (short_answer - 1)" for the first question in the quest with name "Quest 1"
+        """
+        return f"{self.quest.name} ({self.type} - {self.ordinal})"
+
+    @classmethod
+    def next_ordinal(cls, quest):
+        """Return the next ordinal value for a question in this quest"""
+        max_ordinal = Question.objects.filter(quest=quest).aggregate(Max('ordinal'))['ordinal__max']
```

Comment:
This feels overly complicated.  Why not just `Question.objects.filter(quest=quest).last()`? Since we are already setting the order via ordinal property?

Reply:
I'm not sure. I guess my rational was that doing  `Question.objects.filter(quest=quest).aggregate(Max('ordinal'))['ordinal__max']` is getting just a single number from the database. But if  we did `Question.objects.filter(quest=quest).last().ordinal`, I think it's getting the whole object and then the value afterwards?

Performance gains for this are probably negligible though, so maybe `Question.objects.filter(quest=quest).last().ordinal` is better since it is more readable?

Reply:
I doubt there is any performance difference at all.  1. Django will convert the SQL behind the scenes to make it efficient, and both only hit the database once.  2 .You can't get the Max value of a queryset without looking at all objects in the queryset anyway!  

But you got me curious so I asked ChatGPT.  It thinks first() is probably faster for the reasons it gives (grain of salt of course..)
https://chat.openai.com/share/8a774ccf-da5e-45b0-a4d6-8553fbeea62e